dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -156,7 +156,20 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
                         THROW_IF_FAILED(CreateChannelWithRemoteIdHelper(appUserModelId, remoteId, channelInfo));
 
                         // Register a sink with platform which is initialized in the current process
-                        THROW_IF_FAILED(PushNotifications_RegisterNotificationSinkForFullTrustApplication(appUserModelId.get(), PushNotificationManager::Default().as<ABI::Microsoft::Internal::PushNotifications::INotificationListener>().get()));
+                        bool registeredEvent{ false };
+                        {
+                            auto lock{ m_lock.lock_shared() };
+                            registeredEvent = (bool) m_foregroundHandlers;
+                        }
+
+                        if (registeredEvent)
+                        {
+                            {
+                                auto lock{ m_lock.lock_exclusive() };
```

Comment:
Recommend setting this to true only after the RegisterSinkWithPlatform operation below succeeded.